aquadoggo/src/schema/schema_provider.rs

```diff
@@ -114,9 +112,26 @@ impl SchemaProvider {
         Ok(is_update)
     }
 
-    // Return the configured supported schema.
-    pub fn supported_schema_ids(&self) -> Option<&Vec<SchemaId>> {
-        self.supported_schema_ids.as_ref()
+    /// Returns a list of all supported schema ids.
+    ///
+    /// If no whitelist was set it returns the list of all currently known schema ids. If a
+    /// whitelist was set it directly returns the list itself.
+    pub async fn supported_schema_ids(&self) -> Vec<SchemaId> {
+        match &self.whitelisted_schema_ids {
+            Some(schema_ids) => schema_ids.clone(),
+            None => self
+                .all()
+                .await
+                .iter()
+                .map(|schema| schema.id().to_owned())
+                .collect(),
+        }
+    }
+
+    /// Returns true if a whitelist of supported schema ids was provided through user
+    /// configuration.
+    pub fn is_whitelist_active(&self) -> bool {
+        self.whitelisted_schema_ids.is_some()
     }
```

Comment:
It was a little bit hard for me to understand the difference between "supported schema ids" in their different states (with or without config provided), I hope that this API makes it a bit clearer?

Reply:
Yeh, this is better, thanks :+1: